### PR TITLE
shutil.make_archive (py38)

### LINKED ISSUE #4037: shutil.make_archive first argument: _AnyStr vs. _Path/ (_AnyPath)
Hi, running mypy over this in python 3.8:

```python3
path = pathlib.Path.cwd() / "some_folder"
shutil.make_archive(path, "zip", root_dir=path.parent, base_dir=path)
```

This code runs fine, but mypy throws an error related to the first argument. When I [pull up typeshed](https://github.com/python/typeshed/blob/master/stdlib/2and3/shutil.pyi#L133), it seems like the first argument has been annotated as `_AnyStr`. This should probably be changed to `_Path` (at least for 3.8, I haven't checked 3.6/3.7 yet)

### PR DESCRIPTION
**Pull request by zdog234**
fixes #4037 :)

### CODE REVIEW

File: stdlib/2and3/shutil.pyi [hunk 1]
```diff
@@ -130,10 +130,18 @@ if sys.version_info >= (3, 8):
 elif sys.version_info >= (3,):
     def which(cmd: _Path, mode: int = ..., path: Optional[_Path] = ...) -> Optional[str]: ...
 
-def make_archive(base_name: _AnyStr, format: str, root_dir: Optional[_Path] = ...,
-                 base_dir: Optional[_Path] = ..., verbose: bool = ...,
-                 dry_run: bool = ..., owner: Optional[str] = ..., group: Optional[str] = ...,
-                 logger: Optional[Any] = ...) -> _AnyStr: ...
+if sys.version_info >= (3, 8):
+    def make_archive(base_name: _Path, format: str, root_dir: Optional[_Path] = ...,
```
Comment (zdog234) [inline stdlib/2and3/shutil.pyi:134]: I'm not sure whether I should use _Path or _AnyPath

**Comment by hauntsaninja** [comment]
Unfortunately, I don't think we can make this change:
```
Python 3.8.2 (default, Apr 21 2020, 00:39:48) 
[Clang 11.0.0 (clang-1100.0.33.17)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import shutil
>>> from pathlib import Path
>>> shutil.make_archive(Path("abcd"), "zip")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/shantanu/.pyenv/versions/3.8.2/lib/python3.8/shutil.py", line 1045, in make_archive
    filename = func(base_name, base_dir, **kwargs)
  File "/Users/shantanu/.pyenv/versions/3.8.2/lib/python3.8/shutil.py", line 912, in _make_zipfile
    zip_filename = base_name + ".zip"
TypeError: unsupported operand type(s) for +: 'PosixPath' and 'str'
```

It looks like Path only works `base_name` if `root_dir` is not None. If you wanted, you could try and capture that behaviour by using overloads. But I would be careful and check everything, since clearly no one has attempted to ensure that `shutil.make_archive` works well with pathlib.Path

File: stdlib/2and3/shutil.pyi (hunk 1 above)
Comment (hauntsaninja) [reply stdlib/2and3/shutil.pyi:134]: You'd need to use `_AnyPath` if e.g., you required multiple parameters to be the same type, or the return type to be the same type as a parameter.

From a quick scan, this doesn't seem to be the case here.

**Comment by zdog234** [comment]
Ah, I see... Well, it seems like I stumbled into this.

Now that I know about this, I'll probably stay away from using `Path` w/ this function, since it doesn't seem to be be intended behavior.

I think I'll take a deeper look at this though, and check the cpython issue tracker.

Sorry for not catching that before submitting the PR

**Comment by hauntsaninja** [comment]
All good! It's something that you'd expect to work, so definitely worth filing on bugs.python.org

**Pull request closed without merging**